### DISCUSSION TITLE
Fetch Proxy per request

### DISCUSSION THREAD

**Discussion by LiamKarlMitchell** [denoland/deno#15243] [Q&A] [answered]
How can I set an HTTPS proxy to use per each fetch method call.
https://github.com/denoland/deno/issues/3444#issuecomment-1188422838

I would like to have a list of proxies to use rather than just 1 env var as loading them from my own configuration file is easier.

**Comment by zachauten** [comment, marked as answer]
Looks like you can do this with the [createHttpClient](https://doc.deno.land/deno/unstable/~/Deno.createHttpClient) from the unstable api.

**Reply by LiamKarlMitchell** [reply]
Awesome thank you for pointing me in the right direction :)

**Comment by geekyayush** [comment]
Hi, for some reasons, it passes the proxy username and password in `proxy-authorization` header to the URL we are fetching. It's a security concern. Any way to make it not pass that header value?

**Reply by bartlomieju** [reply]
Which version of Deno are you using?

**Reply by geekyayush** [reply]
@bartlomieju 
I am using deno 2.1.7.

**Reply by bartlomieju** [reply]
Please open a report at https://github.com/denoland/deno/security/advisories/new with your reproduction

**Reply by geekyayush** [reply]
Okay, thank you!
